OPCB/functions/missions/gdrunken_spawn.sqf

```diff
@@ -1,38 +1,37 @@
 params ["_pos"];
-_comps = [];
-_group = createGroup east;
-_tanks = [];
+private _comps = [];
+private _tanks = [];
 
 for "_i" from 0 to 3 do {
-	_suitable = [0, 0, 0];
+	_suitable = [0, 0, 0, 0];
 	_maxDist = 1500;
 	_list = [];
 
-	while {((count _suitable) == 3) || {(count _list) > 0}} do {
+	while {count _suitable == 4 || {surfaceIsWater _suitable}} do {
 	
 		_suitable = [_pos, 0, _maxDist, 20, 0, 0.4, 0] call BIS_fnc_findSafePos;
 		
-		if ((count _suitable) == 3) then {
-			_maxDist = worldSize min (_maxDist + 100);
-		} else {
-			_list = nearestTerrainObjects [_suitable,["TREE","BUILDING","RUIN","ROCK","HOUSE"], 55,false];
-		};
+		_maxDist = worldSize min (_maxDist + 100);
+		
+	
+		_list = nearestTerrainObjects [_suitable,[], 55,false];
```

Comment:
If you're not making use of `list` within the loop, then take it out of the loop.

Reply:
fair point